frame/treasury/src/lib.rs

```diff
@@ -1170,93 +346,20 @@ impl<T: Trait<I>, I: Instance> Module<T, I> {
+	// 	T::ModuleId::get().into_sub_account(("bt", id))
+	// }
 
 	/// The needed bond for a proposal whose spend is `value`.
 	fn calculate_bond(value: BalanceOf<T, I>) -> BalanceOf<T, I> {
 		T::ProposalBondMinimum::get().max(T::ProposalBond::get() * value)
 	}
 
-	/// Given a mutable reference to an `OpenTip`, insert the tip into it and check whether it
-	/// closes, if so, then deposit the relevant event and set closing accordingly.
-	///
-	/// `O(T)` and one storage access.
-	fn insert_tip_and_check_closing(
-		tip: &mut OpenTip<T::AccountId, BalanceOf<T, I>, T::BlockNumber, T::Hash>,
-		tipper: T::AccountId,
-		tip_value: BalanceOf<T, I>,
-	) -> bool {
-		match tip.tips.binary_search_by_key(&&tipper, |x| &x.0) {
-			Ok(pos) => tip.tips[pos] = (tipper, tip_value),
-			Err(pos) => tip.tips.insert(pos, (tipper, tip_value)),
-		}
-		Self::retain_active_tips(&mut tip.tips);
-		let threshold = (T::Tippers::count() + 1) / 2;
-		if tip.tips.len() >= threshold && tip.closes.is_none() {
-			tip.closes = Some(system::Module::<T>::block_number() + T::TipCountdown::get());
-			true
-		} else {
-			false
-		}
-	}
-
-	/// Remove any non-members of `Tippers` from a `tips` vector. `O(T)`.
-	fn retain_active_tips(tips: &mut Vec<(T::AccountId, BalanceOf<T, I>)>) {
-		let members = T::Tippers::sorted_members();
-		let mut members_iter = members.iter();
-		let mut member = members_iter.next();
-		tips.retain(|(ref a, _)| loop {
-			match member {
-				None => break false,
-				Some(m) if m > a => break false,
-				Some(m) => {
-					member = members_iter.next();
-					if m < a {
-						continue
-					} else {
-						break true;
-					}
-				}
-			}
-		});
-	}
-
-	/// Execute the payout of a tip.
-	///
-	/// Up to three balance operations.
-	/// Plus `O(T)` (`T` is Tippers length).
-	fn payout_tip(hash: T::Hash, tip: OpenTip<T::AccountId, BalanceOf<T, I>, T::BlockNumber, T::Hash>) {
-		let mut tips = tip.tips;
-		Self::retain_active_tips(&mut tips);
-		tips.sort_by_key(|i| i.1);
-		let treasury = Self::account_id();
-		let max_payout = Self::pot();
-		let mut payout = tips[tips.len() / 2].1.min(max_payout);
-		if !tip.deposit.is_zero() {
-			let _ = T::Currency::unreserve(&tip.finder, tip.deposit);
-		}
-		if tip.finders_fee {
-			if tip.finder != tip.who {
-				// pay out the finder's fee.
-				let finders_fee = T::TipFindersFee::get() * payout;
-				payout -= finders_fee;
-				// this should go through given we checked it's at most the free balance, but still
-				// we only make a best-effort.
-				let _ = T::Currency::transfer(&treasury, &tip.finder, finders_fee, KeepAlive);
-			}
-		}
-		// same as above: best-effort only.
-		let _ = T::Currency::transfer(&treasury, &tip.who, payout, KeepAlive);
-		Self::deposit_event(RawEvent::TipClosed(hash, tip.who, payout));
-	}
-
 	/// Spend some money! returns number of approvals before spend.
-	fn spend_funds() -> Weight {
+	pub fn spend_funds_top() -> Weight {
```

Comment:
Why did you change the name here?

Reply:
Really name change not required, reverted back to "spend_funds()".